### DISCUSSION TITLE
cron & edge function not working locally

### DISCUSSION THREAD

**Discussion by dhurbhat** [supabase/supabase#14747] [Questions]
(reposted here from Discord).
Hi all - I've built an edge function that works locally. Then I added a cron job to invoke this edge function - also locally.  the Job runs and it reports it successful, but the changes that I expect to be made from the edge function are not there.  Some background:

- in running this locally, I created a .env.local file with SUPABASE_URL and ANON_KEY
- the cron job is invoking the local rest API
- tried this with and without running supabase functions serve --env-file ... and w/o expressly running the supabase function serve - both cases the cron job details show SUCCEEDED
- here is the 
```
select
  cron.schedule(
    'run-vote-collator',
    '*/5 * * * *', -- at every 5 minutes
    $$
    select net.http_post(
        url:='http://localhost:54321/functions/v1/collate-votes',
        headers:='{"Authorization": "Bearer MY-TOKEN","Content-Type": "application/json"}'::jsonb,
      ) as request_id;
    $$
  );
```
- the Cron job is scheduled properly
- the job details indicate that the job shows `Success`
- However, there are no changes applied to the db (the edge function makes several updates to the DB).

Questions:

1. can I test the cron job locally or does it have to be run in the cloud?
2. I presume that if I'm testing it locally, the URL has to be available - so am I right in needing to serve up the functions locally?  I've done `supabase functions serve` and `supabase functions serve --env-file .env.local` - to no avail.
3. Where can I look at logs locally (I've checked docker image logs for postgrest and others but there's nothing there of use)? I'm looking to see where the Postgrest SQL API calls are made?
4. is there a way to detect error in the cron job - say if the pg_net failed for some reason??  Is there a way to catch PLSQL errors?

**Comment by dhurbhat** [comment]
Oh! Just got done reading the docs on `pg_net` and learned that I could review the errors for any pg_net calls by issuing the following statement:
```
select * from net._http_response;
```

AND when I do that I get, `Couldn't connect to server`.  

If the local supabase is serving functions as in:
```
sh> supabase functions serve --env-file /Users/sheshadrimantha/SupabaseProjects/supabase/.env.local
Setting up Edge Functions runtime...
Serving functions on http://localhost:54321/functions/v1/<function-name>
```

What could be the problem?
BTW: Once the function is served locally, I can use Insomnia / Postman to make successfully invoke the function.

**Comment by dhurbhat** [comment]
I read somewhere on `pg_net` issues/comments from 2 years ago to modify a ports file (similar to `/etc/hosts`) to allow all on `127.0.0.1` - how would find this config file on a containerized `supabase` local install?

Do I need to change docker and do port forwarding??  Appreciate any help... I'm blocked for now.

**Reply by steve-chavez** [reply]
Check if the discussion here helps: https://github.com/supabase/pg_net/issues/79

**Comment by dhurbhat** [comment]
Amazing thank you so much… not sure how this didn’t come up in my searches
before.
I suspected the docker port … will try.

**Comment by joshm30** [comment]
Cursor AI helped me nail the solution... hopefully this helps

**Option 1:** When using Docker, we need to use host.docker.internal but with the correct format

If you're still getting the "Couldn't resolve host name" error in the Supabase Cron Logs, try these troubleshooting steps:

**Option 2:** Try using the Docker container's internal IP instead of host.docker.internal. Let's get the IP:

**First, verify** 
Docker network settings: there should be a supabase instance
`docker network ls | cat`
Let's check if your Supabase containers are running: there should be multiple supabase containers running
`docker ps | grep supabase | cat`

Try using the Docker container's internal IP instead of host.docker.internal. Let's get the IP:
`docker inspect supabase_kong_re-appeal-vue | grep IPAddress | cat`
My IP was "IPAddress": "172.18.0.5"

Thus
`SELECT net.http_post(
  url := 'http://172.18.0.5:8000/functions/v1/hello-world',
  headers := JSONB_BUILD_OBJECT(
    'Authorization',
    'Bearer auth_code',
    'Content-Type',
    'application/json'
  ),
  body := '{"name": "SQL Test"}',
  timeout_milliseconds := 1000
);`

**More context:**
Since the localhost cron job does not show my edge functions I opted to use a HTTP Request to call my local edge function.  Above is the solution.
